### PR TITLE
Document CNI chaining modes

### PR DESCRIPTION
**Pull request by tgraf**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8874)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-docs-please

File: Documentation/gettingstarted/cni-chaining-calico.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    http://docs.cilium.io
+
+******
+Calico
+******
+
+This guide instructs how to install Cilium in chaining configuration on top of
+Calico:
```
Comment (ianvernon) [inline Documentation/gettingstarted/cni-chaining-calico.rst]: Nit: add a link to Calico here.

File: Documentation/gettingstarted/cni-chaining-generic-veth.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    http://docs.cilium.io
+
+*********************
+Generic Veth Chaining
+*********************
+
+The generic veth chaining plugin enables CNI chaining on top of any CNI plugin
+that is using a veth device model which is the vast majority of plugins.
```
Comment (ianvernon) [inline Documentation/gettingstarted/cni-chaining-generic-veth.rst]: Change to "...veth device model. The majority of CNI plugins use such a model."

File: Documentation/gettingstarted/cni-chaining-generic-veth.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+       103: lxcb3901b7f9c02@if102: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default qlen 1000
+           link/ether 3a:39:92:17:75:6f brd ff:ff:ff:ff:ff:ff link-netnsid 18 promiscuity 0
+           veth addrgenmode eui64 numtxqueues 1 numrxqueues 1 gso_max_size 65536 gso_max_segs 65535
+4. The ``veth`` keyword on line 3 indicates that the network device type is virtual ethernet.
+
+If the CNI plugin you are chaining with is currently not using veth then the
+``generic-veth`` plugin is not suitable. In that case, a full CNI chaining
+plugin is required which understands the device model of the underlying plugin.
+Writing such a plugin is trivial, contact us on Slack for more details.
```
Comment (ianvernon) [inline Documentation/gettingstarted/cni-chaining-generic-veth.rst]: "...is trivial. Contact us on Slack for more details."

Perhaps add a link to the Slack as well.

**Review by ianvernon — CHANGES_REQUESTED** [review]
A few nits.

**Comment by tgraf** [comment]
test-docs-please

**Pull request merged**